app/src/lib/shell.ts

```diff
@@ -0,0 +1,155 @@
+/* tslint:disable:no-sync-functions */
+
+import * as ChildProcess from 'child_process'
+import * as os from 'os'
+
+const ENVIRONMENT_VARIABLES_TO_PRESERVE = new Set([ 'NODE_ENV', 'NODE_PATH' ])
```

Comment:
> Why do we exclude these?

Borrowed from https://github.com/atom/atom/pull/12317. As to the specifics:

- `NODE_PATH` is optional but used for `npm` to resolve modules at additional paths. We shouldn't scrub these if a user has them set, but maybe the values we get from the shell would be more accurate.
- `NODE_ENV` is also optional but for defining `development`/`production` when running scripts - we don't rely on this when running Desktop, but passing it through to user scripts.

> Could we document why?

Sure can, but the more I think about the less I can justify these special cases. Maybe I'm not understanding the audience, but I'd gather these are set in the shell and we should let those flow through here as well.